### PR TITLE
core: sim: ignore overlapping scheduled points

### LINKED ISSUE #7744: Assert check failed on a supposed to be valid train_schedule
### What happened?

We get an assert check failed when loading this train_schedule
```json
{"id":223,"timetable_id":12,"train_name":"failure","labels":[],"rolling_stock_name":"EU4000US","start_time":"2024-06-19T12:00:00Z","path":[{"id":"0","deleted":false,"operational_point":"d9a9cbca-6667-11e3-89ff-01f464e0362d"},{"id":"1","deleted":false,"operational_point":"d97ff4b4-6667-11e3-89ff-01f464e0362d"},{"id":"2","deleted":false,"operational_point":"d9e41972-6667-11e3-89ff-01f464e0362d"},{"id":"3","deleted":false,"operational_point":"d97ff4b4-6667-11e3-89ff-01f464e0362d"},{"id":"4","deleted":false,"operational_point":"d96b10f4-6667-11e3-89ff-01f464e0362d"},{"id":"5","deleted":false,"operational_point":"d94b82c2-6667-11e3-89ff-01f464e0362d"},{"id":"6","deleted":false,"operational_point":"d99b719e-6667-11e3-89ff-01f464e0362d"},{"id":"7","deleted":false,"operational_point":"d9e481f4-6667-11e3-89ff-01f464e0362d"},{"id":"8","deleted":false,"operational_point":"d99b719e-6667-11e3-89ff-01f464e0362d"},{"id":"9","deleted":false,"operational_point":"d98d5cae-6667-11e3-89ff-01f464e0362d"},{"id":"10","deleted":false,"operational_point":"d9834f4c-6667-11e3-89ff-01f464e0362d"},{"id":"11","deleted":false,"operational_point":"d94bb5b4-6667-11e3-89ff-01f464e0362d"},{"id":"12","deleted":false,"operational_point":"d9834f4c-6667-11e3-89ff-01f464e0362d"},{"id":"13","deleted":false,"operational_point":"d9da3a98-6667-11e3-89ff-01f464e0362d"},{"id":"14","deleted":false,"operational_point":"d9e65b56-6667-11e3-89ff-01f464e0362d"}],"schedule":[{"at":"1","arrival":"PT1320S","stop_for":null,"on_stop_signal":false,"locked":false},{"at":"2","arrival":"PT1440S","stop_for":"PT2400S","on_stop_signal":false,"locked":false},{"at":"3","arrival":"PT3900S","stop_for":null,"on_stop_signal":false,"locked":false},{"at":"4","arrival":"PT7320S","stop_for":"PT720S","on_stop_signal":false,"locked":false},{"at":"5","arrival":"PT13320S","stop_for":"PT2760S","on_stop_signal":false,"locked":false},{"at":"6","arrival":"PT21660S","stop_for":null,"on_stop_signal":false,"locked":false},{"at":"7","arrival":"PT21840S","stop_for":"PT8940S","on_stop_signal":false,"locked":false},{"at":"8","arrival":"PT30840S","stop_for":null,"on_stop_signal":false,"locked":false},{"at":"9","arrival":"PT43980S","stop_for":"PT1320S","on_stop_signal":false,"locked":false},{"at":"10","arrival":"PT45840S","stop_for":null,"on_stop_signal":false,"locked":false},{"at":"11","arrival":"PT46020S","stop_for":"PT4980S","on_stop_signal":false,"locked":false},{"at":"12","arrival":"PT51060S","stop_for":null,"on_stop_signal":false,"locked":false},{"at":"13","arrival":"PT58320S","stop_for":"PT2400S","on_stop_signal":false,"locked":false},{"at":"14","arrival":"PT60960S","stop_for":null,"on_stop_signal":false,"locked":false}],"margins":{"boundaries":[],"values":["none"]},"initial_speed":0.0,"comfort":"STANDARD","constraint_distribution":"STANDARD","speed_limit_tag":null,"power_restrictions":[],"options":{"use_electrical_profiles":true}}
```

### What did you expect to happen?

The simulation not to make a hard error

### How can we reproduce it (as minimally and precisely as possible)?

creae a timetable with this train

### What operating system, browser and environment are you using?

.

### OSRD version (top right corner `Account` button > `Informations`)

09eead2

### PR DESCRIPTION
**Pull request by eckter**
Fix https://github.com/OpenRailAssociation/osrd/issues/7744

### CODE REVIEW

**Review by Erashin — APPROVED** [review]

**Review by flomonster — APPROVED** [review]

File: core/src/main/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulation.kt [hunk 1]
```diff
@@ -271,6 +271,11 @@ fun distributeAllowance(
     startOffset: Offset<TravelledPath>,
     endOffset: Offset<TravelledPath>
 ): List<AllowanceRange> {
+    assert(startOffset <= endOffset)
+    if (startOffset == endOffset) {
+        // TODO: raise warning (overlapping scheduled points)
```
Comment (Khoyo) [inline core/src/main/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulation.kt:281]: At least log it?

File: core/src/main/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulation.kt [hunk 2]
```diff
@@ -216,6 +220,7 @@ fun buildFinalEnvelope(
         var extraTime = point.arrival.seconds - arrivalTime
         if (extraTime < 0.0) {
             // TODO: raise a warning
+            standaloneSimLogger.info("Warning: impossible scheduled point")
```
Comment (bougue-pe) [inline core/src/main/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulation.kt]: (nit) Same below:
```suggestion
            standaloneSimLogger.warn("impossible scheduled point")
```

**Review by bougue-pe — APPROVED** [review]
It's OK to merge as-is, but I would make that warning as rare as possible with an early dedup on consecutive identical steps (probably with an info/debug log) to ease the monitoring and separate "legitimate" cases (weird input) from the suspect ones (internal core behavior).

Context: As discussed, this case from the issue is not about consecutive identical step, as they are separated by another separate step on the same block but not at the same place.
So here, the pathfinding is wrongly aggregating identical steps because intermediate ones are on the same block (to be adressed in #7761)

**Comment by eckter** [comment]
> but I would make that warning as rare as possible with an early dedup on consecutive identical steps (probably with an info/debug log) to ease the monitoring and separate "legitimate" cases (weird input) from the suspect ones (internal core behavior).

I've added a simple check to dedup the scheduled items, but I've only merged the ones that are entirely identical (offset, time, stop duration, on stop signal)

File: core/src/main/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulation.kt [hunk 3]
```diff
@@ -271,6 +276,12 @@ fun distributeAllowance(
     startOffset: Offset<TravelledPath>,
     endOffset: Offset<TravelledPath>
 ): List<AllowanceRange> {
+    assert(startOffset <= endOffset)
+    if (startOffset == endOffset) {
+        // TODO: raise warning (overlapping scheduled points)
+        standaloneSimLogger.info("Warning: different scheduled points at the same location")
```
Comment (bougue-pe) [inline core/src/main/kotlin/fr/sncf/osrd/standalone_sim/StandaloneSimulation.kt]: Nit (feel free to resolve and merge): For monitoring, this log is more important than the one on [ScheduleItem parsing](https://github.com/OpenRailAssociation/osrd/pull/7747/files?diff=split&w=1#diff-1d19928711ba9b2048694a68df42df6685708db9697b0fb8e753ae54d3ed3e05R52).

So I would `log.warn` here and downgrade to `log.info` for parsing (or `log.error` here)

**Review by bougue-pe — APPROVED** [review]
Thanks !

**Review by bougue-pe — APPROVED** [review]

**Pull request merged**